src/main.rs

```diff
@@ -146,14 +146,15 @@ fn main() {
         }
 
         let mut xi_config_dir = xi_config.path.clone();
+        // Remove the file name from the config dir string (e.g. /home/rasmus/.config/gxi/preferences.xiconfig -> /home/rasmus/.config/gxi)
         xi_config_dir.truncate(xi_config_dir.rfind("/").unwrap_or_else(|| panic!("{}", gettext("Failed to set config dir!"))));
         core.client_started(&xi_config_dir, include_str!(concat!(env!("OUT_DIR"), "/plugin-dir.in")));
 
         MainWin::new(
             application,
             shared_queue.clone(),
-            &Rc::new(RefCell::new(core.clone())),
-            Arc::new(Mutex::new(xi_config)),
+            Rc::new(RefCell::new(core.clone())),
+            Rc::new(RefCell::new(xi_config)),
```

Comment:
Would this cause problems when we're working with several views?

Reply:
No, I thought that at first too (that's why I had in in a Mutex), but since we don't ever use anything but the mainthread for GTK stuff (so MainWin/PrefsWin/EditView) this is fine. It's _very_ counterintuitive, but we basically set everything up for GTK and then enter the GTK mainloop (on the main thread). At this point everything we do is via the GTK mainloop: GTK polls for new events (everything that's T.connect* is a closure that is triggered by the mainloop) and then calls the appropriate connect_* for the event. This is done sequentially in one thread, so it's guaranteed that this is only used at one place at a time. Can't seem to find the right doc from gtk-rs for this right now, I hope my explanation was somewhat OK :)

Reply:
https://developer.gnome.org/gtk3/stable/gtk3-General.html

Reply:
Ah, that makes perfect sense.